### ISSUE TITLE
TFLite SiLU returns NaN for large negative finite inputs while Keras returns zero

### ISSUE THREAD

**Issue by ALinrunrun** [tensorflow/tensorflow#118529]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

2.21.0

### Custom code

Yes

### OS platform and distribution

Linux Ubuntu 22.04

### Python version

3.11

### Current behavior?

TensorFlow Lite produces different results from Keras for tf.nn.silu on very large negative finite float32 inputs.

For large negative inputs, Keras returns -0.0, which is the expected saturated behavior of x * sigmoid(x). However, the converted TFLite model returns NaN for the same finite inputs.

In the reproducer below, Keras returns -0.0 for -7e29 and -2.5e30, while TFLite returns NaN for both values.

Expected behavior?

The converted TFLite model should match Keras behavior for tf.nn.silu on finite inputs. Very large negative values should saturate to zero instead of returning NaN.

### Standalone code to reproduce the issue

```shell
import os
os.environ.setdefault("TF_CPP_MIN_LOG_LEVEL", "3")

import warnings
warnings.filterwarnings("ignore")

import numpy as np
import tensorflow as tf


@tf.function
def fn(x):
    return tf.nn.silu(x)


x = np.array([-7e29, np.nan, -2.5e30, 1.0, np.inf], dtype=np.float32)

eager = fn(tf.constant(x)).numpy()

cf = fn.get_concrete_function(tf.TensorSpec(x.shape, tf.float32))
tflite_model = tf.lite.TFLiteConverter.from_concrete_functions([cf]).convert()

interpreter = tf.lite.Interpreter(model_content=tflite_model)
interpreter.allocate_tensors()
interpreter.set_tensor(interpreter.get_input_details()[0]["index"], x)
interpreter.invoke()

tflite_out = interpreter.get_tensor(interpreter.get_output_details()[0]["index"])

print(f"input : {x}")
print(f"eager : {eager}")
print(f"tflite: {tflite_out}")

if eager[0] == 0.0 and eager[2] == 0.0 and np.isnan(tflite_out[0]) and np.isnan(tflite_out[2]):
    print("BUG REPRODUCED: TFLite SiLU returns NaN for large negative finite inputs")
else:
    print("not reproduced")
```

### Relevant log output

```shell
input : [-7.0e+29 nan -2.5e+30 1.0e+00 inf]
eager : [-0. nan -0. 0.7310586 inf]
tflite: [ nan nan nan 0.7310586 inf]
BUG REPRODUCED: TFLite SiLU returns NaN for large negative finite inputs
```

**Comment by aravindmurali22** [comment]
Hi @ALinrunrun ,

Thanks for filing the issue. I tried running the reproduction script, but the bug was not reproduced on my side and I am getting the expected output. It looks like this issue is resolved. Could you please try running it again?

logs :
```
input : [-7.0e+29      nan -2.5e+30  1.0e+00      inf]
eager : [-0.                 nan -0.          0.73105854         inf]
tflite: [-0.                 nan -0.          0.73105854         inf]
not reproduced
```

**Comment by ALinrunrun** [comment]
Hi, thanks for checking.

I created a clean conda environment with tf-nightly and re-ran the reproducer. I can still reproduce it.

Environment:
Python 3.11
tf-nightly 2.22.0-dev20260616

Output:
input : [-7.0e+29      nan -2.5e+30  1.0e+00      inf]
eager : [-0.               nan -0.         0.7310586        inf]
tflite: [      nan       nan       nan 0.7310586       inf]

The mismatch I am looking at is for the finite inputs -7e29 and -2.5e30; the NaN input itself is not the concern.

Could you please share the exact TensorFlow version or nightly build you tested with?

**Comment by aravindmurali22** [comment]
Hi @ALinrunrun , thanks for your comment. 

We verified this on our end with TensorFlow 2.21.0 (Python 3.14, NumPy 2.0.2, Keras 3.10.0, Flatbuffers 25.12.19, Absl-py 2.3.1, Requests 2.32.5) and the issue appears to be resolved, as both eager and TFLite outputs now correctly match. Thank you again!
